source/agora/utils/TxBuilder.d

```diff
@@ -552,84 +687,107 @@ unittest
+    assert(tx.outputs.isSorted);
 }
 
 /// Test with one output key
 unittest
 {
-    const result = TxBuilder(GenesisBlock.payments.front)
+    const tx = TxBuilder(GenesisBlock.payments.front)
         .split([WK.Keys.A.address])
         .sign();
 
     // This transaction has 1 txs
-    assert(result.inputs.length == 8);
-    assert(result.outputs.length == 1);
+    assert(tx.inputs.length == 8);
+    assert(tx.outputs.length == 1);
 
-    // 500M
-    const Amount ExpectedAmount = Amount(488_000_000L * 10_000_000L);
-    assert(result.outputs[0].value == ExpectedAmount);
+    const totalInputs = sumOfGenesisFirstTxOutputs();
+    auto implied_fees = totalInputs - tx.outputs[0].value;
+    assert(implied_fees >= defaultFee * tx.sizeInBytes);
 }
 
 /// Test changing the refund address (and merging outputs by extension)
 unittest
 {
-    const result = TxBuilder(GenesisBlock.payments.front)
+    immutable Number = 3;
+    const tx = TxBuilder(GenesisBlock.payments.front)
         // Refund needs to be called first as it resets the outputs
         .refund(WK.Keys.Z.address)
-        .draw(Amount(62_666_666_6666_666L), WK.Keys.byRange.map!(k => k.address).take(3))
+        .draw(Amount(160_000_000_0000_000L), WK.Keys.byRange.map!(k => k.address).take(Number))
         .sign();
 
-    // This transaction has 4 txs (3 targets + 1 refund)
-    assert(result.inputs.length == 8);
-    assert(result.outputs.length == 4);
-
-    assert(result.outputs == [
-        Output(Amount(62_666_666_6666_666L), WK.Keys.A.address),
-        Output(Amount(62_666_666_6666_666L), WK.Keys.C.address),
-        Output(Amount(62_666_666_6666_666L), WK.Keys.D.address),
-        Output(Amount(300_000_000_0000_002L), WK.Keys.Z.address),
+    // This transaction has 4 outputs (3 draw and 1 refund)
+    assert(tx.inputs.length == 8);
+    assert(tx.outputs.length == Number + 1);
+
+    const totalInputs = sumOfGenesisFirstTxOutputs();
+    auto outputs = tx.outputs.map!(o => o.value).reduce!((a,b) => a + b);
+    auto implied_fees = totalInputs - outputs;
+    assert(implied_fees >= defaultFee * tx.sizeInBytes);
+    assert(tx.outputs == [
+        Output(totalInputs - implied_fees - Amount(160_000_000_0000_000L) * 3, WK.Keys.Z.address),
+        Output(Amount(160_000_000_0000_000L), WK.Keys.A.address),
+        Output(Amount(160_000_000_0000_000L), WK.Keys.C.address),
+        Output(Amount(160_000_000_0000_000L), WK.Keys.D.address),
     ].sort.array);
 }
 
 /// Test with a range of tuples
 unittest
 {
     Output[4] outs = [
-        Output(Amount(1_000_000), WK.Keys.A.address),
-        Output(Amount(2_000_000), WK.Keys.C.address),
-        Output(Amount(3_000_000), WK.Keys.D.address),
-        Output(Amount(4_000_000), WK.Keys.E.address),
+        Output(Amount(100000), WK.Keys.A.address),
+        Output(Amount(200000), WK.Keys.C.address),
+        Output(Amount(300000), WK.Keys.D.address),
+        Output(Amount(400000), WK.Keys.E.address),
```

Comment:
Why were these `Amount`s reduced?

Reply:
They are increased to be more than fees required.

Reply:
I was wondering because it is `1_000_000` in the second commit but `100000` in the third.

Reply:
ok I'll revert that change. Wasn't intended.

Reply:
Reverted